webapp/src/dogma/common/components/NewFile.tsx

```diff
@@ -0,0 +1,187 @@
+import { useAppDispatch } from 'dogma/store';
+import Router from 'next/router';
+import { useForm } from 'react-hook-form';
+import { SerializedError } from '@reduxjs/toolkit';
+import { FetchBaseQueryError } from '@reduxjs/toolkit/dist/query';
+import ErrorHandler from 'dogma/features/services/ErrorHandler';
+import Editor, { OnMount } from '@monaco-editor/react';
+import { useRef, useState } from 'react';
+
+const FILE_PATH_PATTERN = /^[0-9A-Za-z](?:[-+_0-9A-Za-z\.]*[0-9A-Za-z])?$/;
+
+type FormData = {
+  name: string;
+  type: 'UPSERT_JSON' | 'UPSERT_TEXT';
+  summary: string;
+  detail: string;
+};
+
+export const NewFile = ({
+  projectName,
+  repoName,
+}: {
+  projectName: string;
+  repoName: string;
+  revision: string;
+}) => {
+  const { colorMode } = useColorMode();
+  const [addNewFle, { isLoading }] = useAddNewFileMutation();
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<FormData>();
+  const dispatch = useAppDispatch();
+  const onSubmit = async (formData: FormData) => {
+    const data = {
+      commitMessage: {
+        summary: formData.summary,
+        detail: formData.detail,
+        markup: markup,
+      },
+      changes: [
+        {
+          path: '/' + formData.name, // TODO: Allow the actual path in the input form i.e. allow slash /
+          type: formData.type,
+          content: editorRef.current.getValue(),
+        },
+      ],
+    };
+    try {
+      const response = await addNewFle({ projectName, repoName, data }).unwrap();
+      if ((response as { error: FetchBaseQueryError | SerializedError }).error) {
+        throw (response as { error: FetchBaseQueryError | SerializedError }).error;
+      }
+      Router.push(`/app/projects/${projectName}/repos/${repoName}/list/head/`);
+      reset();
+      dispatch(
+        createMessage({
+          title: 'New file created',
+          text: `Successfully created ${formData.name}`,
+          type: 'success',
+        }),
+      );
+    } catch (error) {
+      dispatch(
+        createMessage({
+          title: 'Failed to create a new file',
+          text: ErrorHandler.handle(error),
+          type: 'error',
+        }),
+      );
+    }
+  };
+  const [markup, setMarkup] = useState('PLAINTEXT');
+  const editorRef = useRef(null);
+  const handleEditorMount: OnMount = (editor) => {
+    editorRef.current = editor;
+  };
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Flex minWidth="max-content" alignItems="center" mb={4}>
+        <Heading size="lg">Create new file</Heading>
+      </Flex>
+      <Box p={4}>
+        <RadioGroup onChange={setMarkup} value={markup} mb={2}>
+          <Flex gap={2}>
+            <Spacer />
+            <Radio value="PLAINTEXT" colorScheme="teal">
+              plain text
+            </Radio>
+            <Radio value="MARKDOWN" colorScheme="teal">
+              markdown
+            </Radio>
+          </Flex>
+        </RadioGroup>
```

Comment:
This radio button is associated with commit messages. How about moving this around the commit message component?